examples/preprocessing/plot_eeg_csd.py

```diff
@@ -0,0 +1,94 @@
+"""
+=====================================================
+Transform EEG data using current source density (CSD)
+=====================================================
+
+This script shows an example of how to use CSD [1]_ [2]_ [3]_.
+CSD takes the spatial Laplacian of the sensor signal (derivative in both
+x and y). It does what a planar gradiometer does in MEG. Computing these
+spatial derivatives reduces point spread. CSD transformed data have a sharper
+or more distinct topography, reducing the negative impact of volume conduction.
+"""
+# Authors: Alex Rockhill <aprockhill206@gmail.com>
+#
+# License: BSD (3-clause)
+
+# sphinx_gallery_thumbnail_number = 6
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+import mne
+from mne.datasets import sample
+
+print(__doc__)
+
+data_path = sample.data_path()
+
+###############################################################################
+# Load sample subject data
+raw = mne.io.read_raw_fif(data_path + '/MEG/sample/sample_audvis_raw.fif',
+                          preload=True)
+events = mne.find_events(raw)
+raw = raw.pick_types(meg=False, eeg=True, eog=True, ecg=True, stim=False,
+                     exclude=raw.info['bads'])
+raw.set_eeg_reference(projection=True).apply_proj()
+
+###############################################################################
+# Plot the raw data and CSD-transformed raw data
+
+raw.plot()
+raw_csd = mne.preprocessing.compute_current_source_density(raw)
+raw_csd.plot()
+
+###############################################################################
```

Comment:
should maybe add a brief line here like "CSD can also be computed on Evoked (averaged) data. Here we epoch and average the data so we can demonstrate that."